splunk-otel-android/src/main/java/com/splunk/rum/SpanFilterBuilder.java

```diff
@@ -0,0 +1,138 @@
+     *                          rejected.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder rejectSpansByName(Predicate<String> spanNamePredicate) {
+        rejectSpanNamesPredicate = rejectSpanNamesPredicate.or(spanNamePredicate);
+        return this;
+    }
+
+    /**
+     * Remove matching spans from the exporter pipeline.
+     * <p>
+     * Span that contains an attribute with key {@code attributeKey} and value matching the
+     * {@code attributeValuePredicate} will not be exported.
+     *
+     * @param attributeKey            An attribute key to match.
+     * @param attributeValuePredicate A function that returns true if a span containing an attribute
+     *                                with matching value should be rejected.
+     * @return {@code this}.
+     */
+    public <T> SpanFilterBuilder rejectSpansByAttributeValue(
+            AttributeKey<T> attributeKey,
+            Predicate<? super T> attributeValuePredicate) {
+
+        rejectSpanAttributesPredicates.compute(attributeKey,
+                (k, oldValue) -> oldValue == null
+                        ? attributeValuePredicate
+                        : ((Predicate<T>) oldValue).or(attributeValuePredicate));
+        return this;
+    }
+
+    /**
+     * Modify span data before it enters the exporter pipeline.
+     * <p>
+     * An attribute with key {@code attributeKey} and  will be removed from the span before it is exported.
+     *
+     * @param attributeKey An attribute key to match.
+     * @return {@code this}.
+     */
+    public <T> SpanFilterBuilder removeSpanAttribute(AttributeKey<T> attributeKey) {
+        return removeSpanAttribute(attributeKey, value -> true);
+    }
+
+    /**
+     * Modify span data before it enters the exporter pipeline.
+     * <p>
+     * An attribute with key {@code attributeKey} and value matching the
+     * {@code attributeValuePredicate} will be removed from the span before it is exported.
+     *
+     * @param attributeKey            An attribute key to match.
+     * @param attributeValuePredicate A function that returns true if an attribute with matching
+     *                                value should be removed from the span.
+     * @return {@code this}.
+     */
+    public <T> SpanFilterBuilder removeSpanAttribute(
+            AttributeKey<T> attributeKey,
+            Predicate<? super T> attributeValuePredicate) {
+
+        return replaceSpanAttribute(attributeKey, old -> attributeValuePredicate.test(old) ? null : old);
+    }
+
+    /**
+     * Modify span data before it enters the exporter pipeline.
+     * <p>
+     * Value of an attribute with key {@code attributeKey} will be passed to the
+     * {@code attributeValueModifier} function. Value returned by the function will replace the
+     * original value. When the modifier function returns {@code null} the attribute will be removed
```

Comment:
do we have a test for the `null` case?

Reply:
Indirectly - the `removeSpanAttribute()` method passes a function that returns null when the predicate passes.

Reply:
I think it's worth adding a simple direct test to verify that we don't break this with an internal change in the future.